### PR TITLE
Update Libxc and adapt to its new interface

### PR DESCRIPTION
**Pull request by mfherbst**
- [x] Tag and update Libxc.
- [x] Apply and compute for Hartree kernel
- [ ] Test FD Hartree kernel
- [x] Apply for LDA XC kernel
- [x] Test FD LDA kernel
- [ ] GGA XC kernel

### CODE REVIEW

File: src/terms/xc.jl [hunk 1]
```diff
@@ -53,7 +55,31 @@ function ene_ops(term::TermXc, ψ, occ; ρ, kwargs...)
 end
 
 
-# Functionality for building the XC potential term and constructing the builder itself.
+function compute_kernel(term::TermXc; kwargs...)
+    error("Not implemented")
```
Comment (antoine-levitt) [inline src/terms/xc.jl:59]: just `Diagonal(kernel)` no? (for LDA at least)

File: src/terms/xc.jl [hunk 2]
```diff
@@ -116,6 +142,11 @@ sum(V .* dρ) = sum(V .* IF(Xϕ) .* IF(X dϕ))
 and the result follows.
 =#
 
+function max_required_derivative(functional)
+    functional.family == :lda && return 0
+    functional.family == :gga && return 1
+    return -1
```
Comment (antoine-levitt) [inline src/terms/xc.jl:148]: error?

File: src/terms/xc.jl (hunk 1 above)
Comment (mfherbst) [reply src/terms/xc.jl:59]: Yes for LDA it's just what I compute as "kernel" in the apply function. Not yet sure about GGA. I'll look at that tmr. I just wanted to get finite-differences to agree for the first checkpoint ;).

Comment (antoine-levitt) [reply src/terms/xc.jl:59]: GGA once you know what to do for applying the kernel you just have to convert G_to_r calls to G_to_r_matrix and so on. That would also be a nice testcase for forwarddiff. But again if it's too much trouble just forget about it...

**Comment by mfherbst** [comment]
@antoine-levitt Ok I give up on the GGA kernel for now. I included a reference and what I tried to derive it, but the results are not consistent. The corresponding test is marked as broken and the routine returns an error.

**Comment by antoine-levitt** [comment]
Heh OK. Let's merge this and I'll take a look next week. The trick is always to forget about the continuous stuff and simply differentiate at the discrete level.

**Pull request merged**